content/docs/user-guide/visualization/environments/terrain/terrain-developer-guide/future-work.md

```diff
@@ -0,0 +1,55 @@
+---
+linktitle: Future Work
+title: Future Work
+description: Possible future improvements to the terrain system.
+weight: 500
+toc: true
+---
+
+There are a number of improvements that could potentially be made to the terrain system for anyone interested in taking the system further.
+
+## Large-scale improvements
+
+This category of improvements requires architecting new solutions to provide extensive new capabilities.
+
+### 3D geometry support
+
+Instead of a heightfield-based approach, the APIs and renderer could be extended to use a full 3D approach, whether it's through voxels, displacement maps, mesh ingestion, raytracing, or some other technology. This would enable the system to support caves, arches, cliff geometry, and other vertical or overlapping terrain features that simply can't be supported through a pure heightfield approach.
+
+The API extensions would likely consist of redefining the current APIs to provide the _highest elevation_ terrain data that appears below the given query input positions, and then adding more APIs to provide _all_ the terrain data that exists at or below the given query positions.
+
+### GPU calculation support
+
+The terrain system is built on top of the gradient components, which perform all of their calculations CPU-side. However, the types of calculations being performed are ideal for GPU-side calculation. An overhaul or replacement of the gradient system that supports GPU-side calculations could provide orders of magnitude improvements to the terrain system. The primary caveats are that the data is still needed CPU-side to feed to physics, generalized raycasts, and other systems, and that it should be possible to still use the terrain system on devices with limited or no GPU capabilities (low-end phones, headless servers, etc).
+
+## Additional features
+
+This category of improvements should be possible to incrementally add into the system without fundamentally altering anything about how the system currently works.
+
+### Triangle split direction
+
+The terrain currently always triangulates each quad in the same uniform direction. This could be replaced with heuristics for choosing the best direction for each quad. This could even be exposed as multiple heuristics - same direction for every quad, alternating directions per quad, best choice per quad based on vertex heights. This decision should be controlled at the terrain system level, so that the information can be provided consistently to the terrain debugger, terrain physics, terrain rendering, terrain raycasting, etc.
```

Comment:
```suggestion
Currently, the terrain always triangulates each quad in the same uniform direction. This can be replaced with heuristics for choosing the best direction for each quad. This can even be exposed as multiple heuristics: same direction for every quad, alternating directions per quad, or best choice per quad based on vertex heights. This decision should be controlled at the terrain system level, so that the information can be provided consistently to the terrain debugger, terrain physics, terrain rendering, terrain raycasting, etc.
```